### ISSUE TITLE
PUT operation on media item to update caption doesn't work

### ISSUE THREAD

**Issue by tuxpiper** [ushahidi/platform#2017]
### Expected behaviour

The following operation:
```
PUT /api/v3/media/1
Content-type: application/json
Authorization: Bearer xxxxxxxxx
```

with body
```
{ "caption": "new caption" }
```

Should update the caption for the media item

### Actual behaviour

The following response is returned by the server instead

```
500 Internal Server Error
{
  "errors": [
    {
      "status": 0,
      "title": "Validator media.update is not defined",
      "message": "Validator media.update is not defined",
      "class": "Exception",
      "file": "/var/www/html/platform/src/Factory/ValidatorFactory.php",
      "line": 44,
      "trace": [
        {
          "file": "/var/www/html/platform/src/Factory/UsecaseFactory.php",
          "line": 113,
          "function": "get",
          "class": "Ushahidi\\Factory\\ValidatorFactory",
          "type": "->",
          "args": [
            "media",
            "update"
          ]
        },
        {
          "file": "/var/www/html/platform/application/classes/Ushahidi/Rest.php",
          "line": 186,
          "function": "get",
          "class": "Ushahidi\\Factory\\UsecaseFactory",
          "type": "->",
          "args": [
            "media",
            "update"
          ]
        },
        {
          "file": "/var/www/html/platform/vendor/kohana/core/classes/Kohana/Controller.php",
          "line": 84,
          "function": "action_put_index",
          "class": "Ushahidi_Rest",
          "type": "->",
          "args": []
        },
        {
          "function": "execute",
          "class": "Kohana_Controller",
          "type": "->",
          "args": []
        },
        {
          "file": "/var/www/html/platform/vendor/kohana/core/classes/Kohana/Request/Client/Internal.php",
          "line": 97,
          "function": "invoke",
          "class": "ReflectionMethod",
          "type": "->",
          "args": [
            {
              "request": {},
              "response": {}
            }
          ]
        },
        {
          "file": "/var/www/html/platform/vendor/kohana/core/classes/Kohana/Request/Client.php",
          "line": 114,
          "function": "execute_request",
          "class": "Kohana_Request_Client_Internal",
          "type": "->",
          "args": [
            {},
            {}
          ]
        },
        {
          "file": "/var/www/html/platform/vendor/kohana/core/classes/Kohana/Request.php",
          "line": 997,
          "function": "execute",
          "class": "Kohana_Request_Client",
          "type": "->",
          "args": [
            {}
          ]
        },
        {
          "file": "/var/www/html/platform/httpdocs/index.php",
          "line": 38,
          "function": "execute",
          "class": "Kohana_Request",
          "type": "->",
          "args": []
        }
      ]
    }
  ]
}
```

**Comment by jshorland** [comment]
I don't understand the impact of this on the user, and therefore can't prioritize. Moving to icebox.

**Comment by tuxpiper** [comment]
This is technical description of the underlying problem for issue #2018

**Comment by rowasc** [comment]
Merged this with #2018
